src/metabase/driver/generic_sql/native.clj

```diff
@@ -27,9 +27,8 @@
              (.setAutoCommit jdbc-connection false)
              (try
                ;; Set the timezone if applicable
-               (when-let [timezone (driver/report-timezone)]
-                 (when (and (seq timezone)
-                            (contains? (driver/features driver) :set-timezone))
+               (when-let [timezone (:report-timezone settings)]
+                 (when (seq timezone)
```

Comment:
that check still takes place, but it happens when the `:settings` are added to the query in our QP middleware ... https://github.com/metabase/metabase/blob/3aebb0376a16fa32c024cb843e20c70943260b38/src/metabase/driver/query_processor.clj#L116-L119

So the benefit is that redundant checking code like this can be eliminated from each individual driver because the drivers can now simply use the value presented on the query :settings and trust that it's been validated already.  In this case the `(when-let [timezone (:report-timezone settings)]` binding does everything we need because if the timezone exists in the query settings then we should use it, otherwise move on.